d2common/Configuration.go

```diff
@@ -2,54 +2,129 @@ package d2common
 
 import (
 	"encoding/json"
-	"io/ioutil"
-	"log"
+	"fmt"
 	"os"
-	"path"
-	"strings"
+	"path/filepath"
+	"runtime"
 
 	"github.com/mitchellh/go-homedir"
 )
 
-// Configuration defines the configuration for the engine, loaded from config.json
+// Configuration defines the configuration for the engine
 type Configuration struct {
-	Language        string
-	FullScreen      bool
-	Scale           float64
-	RunInBackground bool
-	TicksPerSecond  int
-	FpsCap          int
-	VsyncEnabled    bool
-	MpqPath         string
-	MpqLoadOrder    []string
-	SfxVolume       float64
-	BgmVolume       float64
+	Language          string
+	FullScreen        bool
+	Scale             float64
+	PauseInBackground bool
+	TicksPerSecond    int
+	FpsCap            int
+	VsyncDisabled     bool
+	MpqPath           string
+	MpqLoadOrder      []string
+	SfxVolume         float64
+	BgmVolume         float64
 }
 
-// ConfigBasePath is used for tests to find the base config json file
-var ConfigBasePath = "./"
+const configPath = "config.json"
 
-func LoadConfiguration() *Configuration {
-	configJSON, err := ioutil.ReadFile(path.Join(ConfigBasePath, "config.json"))
-	if err != nil {
-		log.Fatal(err)
-	}
+var defaultLoadOrder = []string{
+	"Patch_D2.mpq",
+	"d2exp.mpq",
+	"d2xmusic.mpq",
+	"d2xtalk.mpq",
+	"d2xvideo.mpq",
+	"d2data.mpq",
+	"d2char.mpq",
+	"d2music.mpq",
+	"d2sfx.mpq",
+	"d2video.mpq",
+	"d2speech.mpq",
+}
+
+var darwinLoadOrder = []string{
+	"Diablo II Patch",
+	"Diablo II Expansion Data",
+	"Diablo II Expansion Music",
+	"Diablo II Expansion Speech",
+	"Diablo II Expansion Movies",
+	"Diablo II Game Data",
+	"Diablo II Speech",
+	"Diablo II Music",
+	"Diablo II Sounds",
+	"Diablo II Movies",
+	"Diablo II Speech",
+}
+
+func LoadConfiguration() (*Configuration, error) {
```

Comment:
For a future revision, this should take a `os.Reader` so that we can easily write a test for this function.  Regardless, we could still write a test that uses the `testdata` directory to load different configs and make sure the defaults apply where necessary.